File: 2661-smallest-missing-non-negative-integer-after-operations/smallest-missing-non-negative-integer-after-operations.cpp
class Solution {
public:
    int findSmallestInteger(vector<int>& nums, int value) {
        int n = nums.size();
        vector<int> arr(value, 0);
        for (int num: nums){
            if (num >= 0) arr[num%value]++;
            else arr[(value-std::abs(num)%value)%value]++;
        }
        int min_reps = n+1;
        int min_idx = -1;
        for (int idx=0; idx<value; idx++){
            if (arr[idx]<min_reps){
                min_reps = arr[idx];
                min_idx = idx;
            }
        }
        return min_reps*value + min_idx;
    }
};

File: 2661-smallest-missing-non-negative-integer-after-operations/smallest-missing-non-negative-integer-after-operations.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        n = len(nums)
        store = [0]*value
        for i in nums:
            if i >= 0:
                store[i%value] += 1
            else:
                store[(value-(abs(i)%value))%value] += 1
        res = 0
        while True:
            if not store[res%value]:
                break
            else:
                store[res%value] -= 1
                res += 1
        return res